**Назирзода Рахимджон**

**Пункт 1**

In [3]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('tips.csv')

In [5]:
df_num = df[['total_bill', 'tip', 'size', 'price_per_person', 'CC Number']]

df_num

total_bill   tip  size  price_per_person         CC Number
0         16.99  1.01     2              8.49  3560325168603410
1         10.34  1.66     3              3.45  4478071379779230
2         21.01  3.50     3              7.00  6011812112971322
3         23.68  3.31     2             11.84  4676137647685994
4         24.59  3.61     4              6.15  4832732618637221
..          ...   ...   ...               ...               ...
239       29.03  5.92     3              9.68  5296068606052842
240       27.18  2.00     2             13.59  3506806155565404
241       22.67  2.00     2             11.34  6011891618747196
242       17.82  1.75     2              8.91     4375220550950
243       18.78  3.00     2              9.39  3511451626698139

[244 rows x 5 columns]

**Пункт 2**

In [7]:
df_num.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 5 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   total_bill        244 non-null    float64
 1   tip               244 non-null    float64
 2   size              244 non-null    int64  
 3   price_per_person  244 non-null    float64
 4   CC Number         244 non-null    int64  
dtypes: float64(3), int64(2)
memory usage: 9.7 KB


In [8]:
df_num.max()

total_bill          5.081000e+01
tip                 1.000000e+01
size                6.000000e+00
price_per_person    2.027000e+01
CC Number           6.596454e+15
dtype: float64

In [9]:
df_num.min()

total_bill          3.070000e+00
tip                 1.000000e+00
size                1.000000e+00
price_per_person    2.880000e+00
CC Number           6.040679e+10
dtype: float64

In [10]:
df_num.describe()

total_bill         tip        size  price_per_person     CC Number
count  244.000000  244.000000  244.000000        244.000000  2.440000e+02
mean    19.785943    2.998279    2.569672          7.888197  2.563496e+15
std      8.902412    1.383638    0.951100          2.914234  2.369340e+15
min      3.070000    1.000000    1.000000          2.880000  6.040679e+10
25%     13.347500    2.000000    2.000000          5.800000  3.040731e+13
50%     17.795000    2.900000    2.000000          7.255000  3.525318e+15
75%     24.127500    3.562500    3.000000          9.390000  4.553675e+15
max     50.810000   10.000000    6.000000         20.270000  6.596454e+15

**total_bill**
- Непустые значения: 244
- Тип данных: float
- Max: 50.81
- Min: 3.07
- Столбец соержит общий счет стола в долларах

**tip**
- Непустые значения: 244
- Тип данных: float
- Max: 10.00
- Min: 1.00
- Общий чаевые оставленные клиентом

**size**
- Непустые значения: 244
- Тип данных: int
- Max: 6.00
- Min: 1.00
- Общий размер группы клиентов

**price_per_person**
- Непустые значения: 244
- Тип данных: float
- Max: 20.27
- Min: 2.88
- Средняя стоимость на одного человека

**СС Number**
- Непустые значения: 244
- Тип данных: int
- Max: 6.59
- Min: 6.04
- Номер кредитной карты

**Пункт 3**

In [33]:
df_num

total_bill  tip  size  price_per_person         CC Number  after_dot  \
0          17.0  1.0     2              8.49  3560325168603410        0.0   
1          10.0  2.0     3              3.45  4478071379779230        0.0   
2          21.0  4.0     3              7.00  6011812112971322        0.0   
3          24.0  3.0     2             11.84  4676137647685994        0.0   
4          25.0  4.0     4              6.15  4832732618637221        0.0   
..          ...  ...   ...               ...               ...        ...   
239        29.0  6.0     3              9.68  5296068606052842        0.0   
240        27.0  2.0     2             13.59  3506806155565404        0.0   
241        23.0  2.0     2             11.34  6011891618747196        0.0   
242        18.0  2.0     2              8.91     4375220550950        0.0   
243        19.0  3.0     2              9.39  3511451626698139        0.0   

     percent  sum_bill_tip  
0       5.88          18.0  
1      20.00          12.0  
2      19.05          25.0  
3      12.50          27.0  
4      16.00          29.0  
..       ...           ...  
239    20.69          35.0  
240     7.41          29.0  
241     8.70          25.0  
242    11.11          20.0  
243    15.79          22.0  

[244 rows x 8 columns]

In [31]:
df_num = df_num.copy()

df_num["after_dot"] = df_num["total_bill"] % 1

df_num["total_bill"] = df_num["total_bill"].round()

df_num["tip"] = df_num["tip"].round()

df_num["percent"] = (df_num["tip"] / df_num["total_bill"] * 100).round(2)

df_num["sum_bill_tip"] = df_num["total_bill"] + df_num["tip"]

**Пункт 4**

In [50]:
df_str = df[[
    'sex',
    'smoker',
    'day',
    'time',
    'Payer Name']]

df_str

sex smoker   day    time          Payer Name
0    Female     No   Sun  Dinner  Christy Cunningham
1      Male     No   Sun  Dinner      Douglas Tucker
2      Male     No   Sun  Dinner      Travis Walters
3      Male     No   Sun  Dinner    Nathaniel Harris
4    Female     No   Sun  Dinner        Tonya Carter
..      ...    ...   ...     ...                 ...
239    Male     No   Sat  Dinner       Michael Avila
240  Female    Yes   Sat  Dinner      Monica Sanders
241    Male    Yes   Sat  Dinner          Keith Wong
242    Male     No   Sat  Dinner        Dennis Dixon
243  Female     No  Thur  Dinner     Michelle Hardin

[244 rows x 5 columns]